src/fsharp/LanguageFeatures.fs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All Rights Reserved. See License.txt in the project root for license information.
+
+/// Coordinating compiler operations - configuration, loading initial context, reporting errors etc.
+module internal FSharp.Compiler.Features
+
+open System
+
+//------------------------------------------------------------------------------------------------------------------
+// Language version command line switch
+//------------------------------------------------------------------------------------------------------------------
+// Add your features to this List - in code use languageVersion.SupportsFeature(LanguageFeatures.yourFeature) 
+// a return value of false means your feature is not supported by the user's language selection
+// All new language features added from now on must be protected by this.
+// Note:
+//   *  The fslang design process will require a decision about feature name and whether it is required.
+//   *  When a feature is assigned a release language, we will scrub the code of feature references and apply
+//      the Release Language version.
+
+/// LanguageFeature enumeration
+[<RequireQualifiedAccess>]
+type LanguageFeature =
+    | LanguageVersion46 = 0
+    | LanguageVersion47 = 1
+    | SingleUnderscorePattern = 2
+    | Nullness = 1000
+    | ScriptingPackageManagement = 1001
+
+/// LanguageVersion management
+type LanguageVersion (specifiedVersion) =
+
+    // When we increment language versions here preview is higher than current RTM version
+    static let languageVersion46 = 4.6m
```

Comment:
I don't think that using numbers is better than enum here.

Reply:
Well we do a numeric comparison on the number, to figure out if the version is supported or not.  Sure we could compare enum ordinals, but it's not really much of a thing.